tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,495 @@
+            let _ = pubfile.write_all(text.as_bytes());
+        }
+    }
+
+    #[test]
+    fn test002_no_keys() {
+        let aks = open_authorized_keys();
+
+        aks.write().expect("failed to update authorized keys directory");
+
+        let authkeysdir: PathBuf = PathBuf::from(format!("{}/authorized_keys.d", SSH_DIR));
+        assert!(authkeysdir.is_dir());
+
+        let authkeys: PathBuf = PathBuf::from(format!("{}/authorized_keys", SSH_DIR));
+        assert!(!authkeys.exists());
+    }
+
+    #[test]
+    fn test003_first_run() {
+        // It is necessary to clean up before running the first run test,
+        // because old_authorized_keys will be only created when authorized_keys.d
+        // does not exist.
+        _ = fs::remove_dir_all(format!("{}/authorized_keys.d", SSH_DIR));
+
+        let authkeys: PathBuf = PathBuf::from(format!("{}/authorized_keys", SSH_DIR));
+        let mut authkeysfile =
+            File::options().create(true).append(true).write(true).open(authkeys.clone()).expect(format!("unable to create a file {}", authkeys.to_str().unwrap_or_default()).as_str());
+        for (_, text) in &*TEST_KEYS {
+            _ = authkeysfile.write_all(format!("{}\n", text).as_bytes());
+        }
+
+        // equivalent of running "update-ssh-keys" without args
+        let aks = open_authorized_keys();
+
+        aks.write().expect("failed to update authorized keys directory");
+        aks.sync().expect("failed to update authorized keys");
+
+        let authkeys: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/old_authorized_keys", SSH_DIR));
+        assert!(authkeys.exists());
+        assert_has_keys(["valid1", "valid2", "valid3"].to_vec());
+    }
+
+    #[test]
+    fn test004_add_one_file() {
+        // "update-ssh-keys --add one valid1.pub"
+        add_one_ssh_key(
+            "valid1",            // pubkeyname
+            "one",               // testname
+            false,               // is_force
+            true,                // is_replace
+            false,               // is_stdin
+            true,                // is_expected_success
+            ["valid1"].to_vec(), // assert_keys
+        );
+    }
+
+    #[test]
+    #[ignore] // ignore the test, as the test hangs forever.
+    fn test005_add_one_stdin() {
+        // "update-ssh-keys --add one valid1.pub"
+        add_one_ssh_key(
+            "valid1",            // pubkeyname
+            "one",               // testname
+            false,               // is_force
+            false,               // is_replace
+            true,                // is_stdin
+            true,                // is_expected_success
+            ["valid1"].to_vec(), // assert_keys
+        );
+    }
+
+    #[test]
+    fn test006_replace_one() {
+        // "update-ssh-keys --add one valid1.pub"
+        test004_add_one_file();
```

Comment:
Since each test needs its own setup and teardown, I suggest not to call it here but copy the contents over